clipmenu

```diff
@@ -40,32 +40,26 @@ fi
 # rofi supports dmenu-like arguments through the -dmenu flag
 [[ "$CM_LAUNCHER" == rofi ]] && set -- -dmenu "$@"
 
-list_clips() {
-    LC_ALL=C sort -rnk 1 < "$cache_file" | cut -d' ' -f2- | awk '!seen[$0]++'
-}
-
 if [[ "$CM_LAUNCHER" == rofi-script ]]; then
     if (( $# )); then
         chosen_line="${!#}"
     else
-        list_clips
+        clipcli list
         exit
     fi
 else
-    chosen_line=$(list_clips | "$CM_LAUNCHER" "${launcher_args[@]}" "$@")
-    launcher_exit=$?
+    chosen_line=$(clipcli list | "$CM_LAUNCHER" "${launcher_args[@]}" "$@")
 fi
 
 [[ $chosen_line ]] || exit 1
-file=$cache_dir/$(cksum <<< "$chosen_line")
-[[ -f "$file" ]] || exit 2
+clip(){ clipcli get "$chosen_line"; }
```

Comment:
`$chosen_line` comes implicitly, which makes this code hard to reason about for humans or static analysis. Just open code it.